source/grammar.tex

```diff
@@ -18,7 +18,7 @@
 
 \pnum
 \indextext{keyword}%
-New context-dependent keywords are introduced into a program by
+New names are introduced into a program by
```

Comment:
And function and variable declaration are different? In what respect?

I think this entire paragraph should just be removed.

@zygoloid , do you have an idea what this does here?

Reply:
I think this is trying to describe the so-called lexer hack, whereby the grammatical interpretation of the program depends on the result of name lookup. I don't think this really gets there for me, though, and it'd be more useful to say that an *identifier* is only a *foo-name* if name lookup finds a foo, analogous to [this](https://eel.is/c++draft/expr#prim.id.unqual-1).

Reply:
@Eisenwave Do you want to try that out?

Reply:
This would be better left to someone who understands this lexer hack better. I'm not confident that I can get the wording right.

Reply:
@zygoloid , we already say that the _identifier_ in a class-head introduces a class-name. We might lack a bit of wording for _template-name_, but that's a core issue.